### PR TITLE
fix: bugfix for channel anchor

### PR DESCRIPTION
**Pull request by nuanyang233**
# Pull Request

## Related Issue

Closes FW-1090

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/components/Markup/MarkupLink/ChannelTag.tsx [hunk 1]
```diff
@@ -56,6 +56,8 @@ export const ChannelTag = memo<Omit<MarkupLinkProps, 'post'>>(function ChannelTa
 
     if (!channelId) return;
 
+    if (channelId.includes('/')) return channelId;
```
Comment (guanbinrui) [inline src/components/Markup/MarkupLink/ChannelTag.tsx:59]:
```suggestion
    if (channelId.startsWith('/')) return channelId;
```

File: src/constants/regexp.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const MENTION_REGEX = /@[^\s()@:%+~#?&=,!?']+/g;
 
 export const HASHTAG_REGEX = /(#\w*[A-Za-z]\w*)/g;
 
-export const CHANNEL_REGEX = /(^|\s)\/[a-z0-9-]+/g;
+export const CHANNEL_REGEX = /(^|\s)\/[a-z0-9-/]+/g;
```
Comment (guanbinrui) [inline src/constants/regexp.ts:11]: Please add unit testing.

File: src/components/Markup/MarkupLink/ChannelTag.tsx (hunk 1 above)
Comment (nuanyang233) [reply src/components/Markup/MarkupLink/ChannelTag.tsx:59]: No, this is to distinguish dates such as `/2024/05/21`.

Comment (guanbinrui) [reply src/components/Markup/MarkupLink/ChannelTag.tsx:59]: You can eliminate it in the regexp.

File: src/constants/regexp.ts (hunk 2 above)
Comment (guanbinrui) [inline src/constants/regexp.ts:11]:
```suggestion
export const CHANNEL_REGEX = /(^|\s)\/[a-z0-9-]+($|(?![0-9a-z\/]))/g;
```

**Pull request merged**